src/EditModel/event-program/assign-data-elements/AssignDataElements.js

```diff
@@ -224,12 +236,28 @@ function AssignDataElements(props, { d2 }) {
     const itemStore = Store.create();
     const assignedItemStore = Store.create();
 
+    //Fix for DHIS2-4369 where some program stages may contain other dataelements than TRACKER
+    //This is due to a database inconsistency. This fix makes it possible to show and be able to remove these
+    //elements from the UI
+    const otherElems = props.model.programStageDataElements
+    .filter(v => {
+        const { dataElement } = v;
+        return dataElement.domainType && dataElement.domainType !== "TRACKER";
```

Comment:
Maybe I am missing something here, but this condition does not seem to match the comment above completely:
- From the comment I conclude that you want exclude all dataElements with domainType TRACKER, and allow all others.
- In the condition you are explicitly excluding:
   a) dataElements without a domainType
   b) dataElements with domainType TRACKER
If you only want to exclude dataElements with domainType TRACKER, then you should remove the first part of the condition. `undefined !== 'TRACKER'` will evaluate to `true` and I think that is what you want.

Quite likely this will never cause an actual problem, because dataElements always have a domainType (?). But it would improve readability if you remove the condition anyway.

Reply:
The idea was to exclude dataElement without domainType - just as a safeguard, but it might be unnecessary. Thanks for you review.

Reply:
It could lead to unexpected behavior. Thanks for removing it.